http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala

```diff
@@ -93,6 +110,11 @@ class AzurePubsubHandlerInterp[F[_]: Parallel](
         )
       )
       createVmJobId = WsmJobId(s"create-vm-${runtime.id.toString.take(10)}")
+
```

Comment:
I know this isn't technically a part of the review, but ...
seeing how as our service is making requests in a polling nature to a shared resource, does it make sense to add a `leo-` prefix to the jobId? or is there another way that we are keeping track of leo-based job requests?

Reply:
Hmm.. it might be worth doing this in a larger PR to add this to all leo requests so we can retrieve all leo-related jobs.

We don't store job IDs at all, so if leo restarts mid job polling, we lose this ID entirely I believe. cc @lucymcnatt , we should perhaps think in a larger sense how we can handle this and not end up with "stuck" resources, unless I'm missing something.

Reply:
> ... if leo restarts mid job polling, we lose this ID entirely I believe
this kind of defeats the purpose and intention of the caller creating and submitting a job-id to an async endpoint. how hard would it be to take the job-id and storing that on our end to give a reference point to "check"

cc: @dvoet

Reply:
Yeah, the issue comes back to leo and WSM are both storing this state...

Almost seems like we should nix leo internal state entirely on these objects, have it be a nullable field in our DB now for backwards compatibility, and if its null look it up in WSM 🤔   

Then we wouldn't have to poll at all... just fire and forget, and look up the resource in WSM whenever we need to report state to an end user request

Reply:
> Yeah, the issue comes back to leo and WSM are both storing this state...
For me, this comes down to the fact we are aiming to leverage an async endpoint, but are consequently throwing away the job-id we provided to that async endpoint to track that job's overall progress (and therefore no longer able to track the async job's state).

That being said, are things idempotent in the currently designed solution? or do we stand to create orphaned resources?

Reply:
yeah there's definitely an opportunity for some work to be done to make the polling resources idempotent...I do like the idea of standardizing the job id names in some way, then we can always grab the id with the action + wsmResourceId or something. I don't know if its necessary for it to include a leo-specific prefix since leo is the only service contacting WSM to create/delete runtimes AFAIK

Reply:
on your thought @jdcanas on not polling altogether....it's definitely a tempting idea, but that would also mean with every get request from the UI we'd have to send an extra WSM get request for the state. Which would work, but maybe is kinda cumbersome and adds extra load on WSM. This problem is also why I like the idea Liz proposed in our walkthrough last week, of making the new services handle the full life-cycle of said resource --> ie we wouldn't have to call WSM anymore for AzureControlledResources.

Reply:
Hmmmm.... maybe a way to get the best of both worlds is to pop createDisk out to a higher level and not to it at all as part of `AzurePubsubHhandler`'s `createAndPollRuntime` which is the interface exposed. Then, the caller of `AzurePubsub` can instead decide whether to create a disk first, and pass that to the VM creation if needed.

Reply:
I made a ticket for chip's concerns though https://broadworkbench.atlassian.net/browse/IA-4850